roles/openshift_health_checker/library/aos_version.py

```diff
@@ -52,32 +55,41 @@ def main():
         module.fail_json(msg="aos_version module could not import yum: %s" % IMPORT_EXCEPTION)
 
     # determine the packages we will look for
-    rpm_prefix = module.params['rpm_prefix']
-    if not rpm_prefix:
-        module.fail_json(msg="rpm_prefix must not be empty")
-    expected_pkgs = set([
-        rpm_prefix,
-        rpm_prefix + '-master',
-        rpm_prefix + '-node',
-    ])
-
-    # determine what level of precision the user specified for the openshift version.
-    # should look like a version string with possibly many segments e.g. "3.4.1":
-    requested_openshift_release = module.params['requested_openshift_release']
+    package_list = module.params['package_list']
+    if not package_list:
+        module.fail_json(msg="package_list must not be empty")
+
+    # generate set with only the names of expected packages
+    expected_pkgs = set(map(lambda p: p[LIST_PKG_NAME], package_list))
+
+    # gather packages that require a multi_minor_release check
+    multi_minor_pkgs = filter(lambda p: p[LIST_PKG_MULTI_MINOR_CHECK], package_list)
+
+    # generate list of packages with a specified (non-empty) version
+    # should look like a version string with possibly many segments e.g. "3.4.1"
+    versioned_pkgs = filter(lambda p: p[LIST_PKG_VERSION], package_list)
 
     # get the list of packages available and complain if anything is wrong
     try:
         pkgs = _retrieve_available_packages(expected_pkgs)
-        if requested_openshift_release:
-            _check_precise_version_found(pkgs, expected_pkgs, requested_openshift_release)
-            _check_higher_version_found(pkgs, expected_pkgs, requested_openshift_release)
-        if module.params['openshift_deployment_type'] in ['openshift-enterprise']:
-            _check_multi_minor_release(pkgs, expected_pkgs)
+        if versioned_pkgs:
+            _check_precise_version_found(pkgs, _to_dict(versioned_pkgs))
+            _check_higher_version_found(pkgs, _to_dict(versioned_pkgs))
+        if multi_minor_pkgs:
+            _check_multi_minor_release(pkgs, _to_dict(multi_minor_pkgs))
     except AosVersionException as excinfo:
         module.fail_json(msg=str(excinfo))
     module.exit_json(changed=False)
 
 
+def _to_dict(pkg_list):
+    d = dict()
+    for pkg in pkg_list:
+        d[pkg[LIST_PKG_NAME]] = pkg
+
+    return d
```

Comment:
Concise and clearer as a comprehension...

    return {pkg[LIST_PKG_NAME]: pkg for pkg in pkg_list}